applications/GeoMechanicsApplication/custom_elements/compressibility_calculator.h

```diff
@@ -0,0 +1,98 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#pragma once
+
+#include "calculator.h"
+#include "custom_retention/retention_law.h"
+#include "includes/properties.h"
+#include "includes/ublas_interface.h"
+
+#include <utility>
+#include <vector>
+
+namespace Kratos
+{
+
+class CompressibilityCalculator : public Calculator
+{
+public:
+    class InputProvider
+    {
+    public:
+        InputProvider(std::function<const Properties&()>                         rElementProperties,
+                      std::function<const std::vector<RetentionLaw::Pointer>&()> rRetentionLaws,
+                      std::function<const Matrix&()>                             rNContainer,
+                      std::function<Vector()>                        rIntegrationCoefficients,
+                      std::function<double()>                        DtPressureCoefficient,
+                      std::function<Vector(const Variable<double>&)> rNodalValuesOfDtWaterPressure)
```

Comment:
Also here, I would suggest to add `Get` to the function parameter names (see my previous comment about `PermeabilityCalculator::InputPrivider`):
```suggestion
        InputProvider(std::function<const Properties&()>                         GetElementProperties,
                      std::function<const std::vector<RetentionLaw::Pointer>&()> GetRetentionLaws,
                      std::function<const Matrix&()>                             GetNContainer,
                      std::function<Vector()>                        GetIntegrationCoefficients,
                      std::function<double()>                        GetDtPressureCoefficient,
                      std::function<Vector(const Variable<double>&)> GetNodalValuesOfDtWaterPressure)
```

Since we use pass-by-value, we don't need the `r` prefix.

Reply:
Done